soroban-env-host/src/host/declared_size.rs

```diff
@@ -142,6 +145,9 @@ impl_declared_size_type!(ScString, 24);
 impl_declared_size_type!(ScSymbol, 24);
 impl_declared_size_type!(CreateContractArgs, 99);
 impl_declared_size_type!(ContractIdPreimage, 66);
+impl_declared_size_type!(ContractDataType, 4);
+impl_declared_size_type!(ContractLedgerEntryType, 66);
+impl_declared_size_type!(ContractDataEntry, 128);
```

Comment:
@jayz22 is this fine? Or is there an issue with declaring ContractDataEntry and LedgerKeyContractData a declared size type when they contain ScVals?

Reply:
We've discussed offline, for types with deep structures, we need to manually implement `MeteredClone` and `Compare` that calls into the substructures for meter charging. Otherwise we will just be charging for a shallow clone/cmp and underestimate the object size. Here we do not need the `declared_size` for `ContractDataEntry`, since we are breaking apart its substructures. 

I've added a couple of commits to your branch (fixing this and some other loose ends)  https://github.com/sisuresh/rs-soroban-env/pull/6